wandb/sdk/internal/internal_api.py

```diff
@@ -1526,6 +1531,11 @@ def upsert_run(
         if num_retries is not None:
             kwargs["num_retries"] = num_retries
 
+        if (project or util.auto_project_name(program_path)) == "model-registry":
```

Comment:
This won't work.  Raising an error like this will crash the program and not clearly communicate to the user what happened.  

We should probably just get rid of the changes in internal_api and add logic to settings.  [This is probably](https://github.com/wandb/wandb/blob/main/wandb/sdk/wandb_settings.py#L734) where we should add it.  We may as well add a constant with reserved project names there that [match these](https://github.com/wandb/core/blob/master/services/gorilla/api/resolver/query.go#L530).

Curiously gorilla currently allows projects named `model-registry` and `model-registry2` so if you really do want to reserve those we should update that file.  I imagine we actually want to reserve `registry`.  @KyleGoyette [has a PR](https://github.com/wandb/core/pull/11999/files) to reserve `launch` as well.

Reply:
I see, we should also stop `registry` and `launch` along with `model-registry`.  As far as making the backend prevent project creation we should definitely do that as well.  I'm not sure how the `model-registry` get's created currently, but I would imagine it doesn't go through the resolvers.  Even if it does, we can just add an additional input parameter that says "I know what I'm doing, let me create this"...

Reply:
Yeah, right now only the creation of `registry` is blocked on the backend (due to a url conflict). I hadn't blocked `model-registry` as that still had to be allowed, just not when the user did it manually. It's required to be there as more of a hidden project for all registered models. Right now, it's created via the `LinkArtifact` [resolver](https://github.com/wandb/core/blob/c723ba81bfbf98d8c4010ea4731a2137393485fb/services/gorilla/api/resolver/mutation.go#L10122) => `createArtifactPortfolio` => `getProjectFromNameOrCreate`.  

If I were to bypass the block to create `model-registry` in `getProjectFromNameOrCreate`, it would still allow users to initialize runs and log artifacts in there.

Reply:
I'm thinking I could potentially move [this block ](https://github.com/wandb/core/blob/c723ba81bfbf98d8c4010ea4731a2137393485fb/services/gorilla/api/resolver/mutation.go#L720-L725) from within `getProjectFromNameOrCreate` to the functions that make the call (currently there are only 3 such functions). That way I can control when the creation of the `model-registry` project is blocked. This would avoid having that error path in the public api as well.

Reply:
Yeah, either that or add an argument to `getProjectFromNameOrCreate` that only the `createArtifactPortfolio` sets to true that bypasses the restricted project check.

Reply:
Great, thanks!